src/events/on_message_auto_thread.ts

```diff
@@ -52,14 +50,42 @@ function get_thread_name(message: Message): string | Promise<string> {
 }
 
 function instruction_message(thread: ThreadChannel): MessageOptions {
+	const base_description =
+		"I've created a thread for your message. Please continue any relevant discussion in this thread. You can rename it with the `/thread rename` command if I failed to set a proper name for it.";
+
+	if (!HELP_CHANNELS.includes(thread.parentId!))
+		return {
+			embeds: [
+				build_embed({
+					description: base_description,
+				}),
+			],
+		};
+
+	const archiveButton = new MessageButton({
+		label: 'Archive',
+		customId: 'thread-archive',
+		emoji: '🔒',
+		style: 'SECONDARY',
+	});
+
+	const solveButton = new MessageButton({
+		label: 'Solved',
+		customId: 'thread-solved',
+		emoji: '✅',
+		style: 'PRIMARY',
+	});
+
+	const row = new MessageActionRow({
+		components: [solveButton, archiveButton],
+	});
+
 	return {
+		components: [row],
+
 		embeds: [
 			build_embed({
-				description: `I've created a thread for your message. Please continue any relevant discussion in this thread. You can rename it with the \`/thread rename\` command if I failed to set a proper name for it.${
-					HELP_CHANNELS.includes(thread.parentId!)
-						? '\nWhen your problem is solved you can archive it with `/thread archive`'
-						: ''
-				}`,
+				description: `${base_description}\n\nWhen your problem is solved you can archive it with \`/thread archive\`\n\nWhen your problem is solved run \`/thread solve\`, don't forget to credit the person that helped you!`,
```

Comment:
```suggestion
				description: `${base_description}\n\nWhen your problem is solved run \`/thread solve\`, don't forget to credit the person that helped you!`,
```

Should just encourage marking them as solved and reduce verbosity. Cases with people wanting to archive it without marking it as solved would likely be rare.